File: java/src/PaintCalculator.java

public class PaintCalculator {

	private static final int PAINT_FOR_CEILING = 350;
	Room room;

	public PaintCalculator(Room room) {
		this.room = room;
	}
	
	private double calculateArea(){
		return room.getArea();
		
	}

	public int getGallonsOfPaintForCeiling() {
      double gallonsNeeded = calculateArea() / PAINT_FOR_CEILING;
      int result = (int) Math.ceil(gallonsNeeded);
      return result;
	}

}

File: ruby/lib/paint_calculator.rb
class PaintCalculator
  PAINT_FOR_CEILING = 350

  def initialize(room)
    @room = room
  end

  
  def gallons_of_paint_for_ceiling
    result = calculate_area / PAINT_FOR_CEILING
    result.ceil
  end

  private
    def calculate_area
      @room.area
    end

end
